src/pages-and-resources/discussions/app-list/AppList.test.jsx

```diff
@@ -20,84 +19,109 @@ import { fetchDiscussionSettings, fetchProviders } from '../data/thunks';
+      });
 
-  function createComponent(screenWidth = breakpoints.extraLarge.minWidth) {
-    return (
-      <AppProvider store={store}>
-        <ResponsiveContext.Provider value={{ width: screenWidth }}>
-          <IntlProvider locale="en" messages={{}}>
-            <AppList />
-          </IntlProvider>
-        </ResponsiveContext.Provider>
-      </AppProvider>
-    );
-  }
+      store = await initializeStore();
+      axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+      await mockStore(piazzaApiResponse);
+    });
 
-  beforeEach(async () => {
-    initializeMockApp({
-      authenticatedUser: {
-        userId: 3,
-        username: 'abc123',
-        administrator: true,
-        roles: [],
-      },
+    test('display a card for each available app', async () => {
+      renderComponent();
+      const appCount = store.getState().discussions.appIds.length;
+      expect(screen.queryAllByRole('radio')).toHaveLength(appCount);
     });
 
-    store = await initializeStore();
-    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
-  });
+    test('displays the FeaturesTable at desktop sizes', async () => {
+      renderComponent();
+      expect(screen.queryByRole('table')).toBeInTheDocument();
+    });
 
-  const mockStore = async (mockResponse, screenWidth = breakpoints.extraLarge.minWidth) => {
-    axiosMock.onGet(getDiscussionsProvidersUrl(courseId)).reply(200, generateProvidersApiResponse());
-    axiosMock.onGet(getDiscussionsSettingsUrl(courseId)).reply(200, mockResponse);
-    await executeThunk(fetchProviders(courseId), store.dispatch);
-    await executeThunk(fetchDiscussionSettings(courseId), store.dispatch);
-    const component = createComponent(screenWidth);
-    const wrapper = render(component);
-    container = wrapper.container;
-  };
+    test('hides the FeaturesTable at mobile sizes', async () => {
+      renderComponent(breakpoints.extraSmall.maxWidth);
+      expect(screen.queryByRole('table')).not.toBeInTheDocument();
+    });
 
-  test('display a card for each available app', async () => {
-    await mockStore(piazzaApiResponse);
-    const appCount = store.getState().discussions.appIds.length;
-    expect(queryAllByRole(container, 'radio')).toHaveLength(appCount);
-  });
+    test('hides the FeaturesList at desktop sizes', async () => {
+      renderComponent();
+      expect(screen.queryByText(messages['supportedFeatureList-mobile-show'].defaultMessage)).not.toBeInTheDocument();
+    });
 
-  test('displays the FeaturesTable at desktop sizes', async () => {
-    await mockStore(piazzaApiResponse);
-    expect(queryByRole(container, 'table')).toBeInTheDocument();
-  });
+    test('displays the FeaturesList at mobile sizes', async () => {
+      renderComponent(breakpoints.extraSmall.maxWidth);
+      const appCount = store.getState().discussions.appIds.length;
+      expect(screen.queryAllByText(messages['supportedFeatureList-mobile-show'].defaultMessage)).toHaveLength(appCount);
```

Comment:
You can directly pass the length instead of creating a variable appCount. It's recommended to avoid declaring variables that are only used once for the sake of cleaner and more concise code.